src/shadowbox/model/errors.ts

```diff
@@ -24,15 +24,14 @@ class OutlineError extends Error {
 }
 
 export class InvalidPortNumber extends OutlineError {
-  // Since this is the error when a non-numeric value is passed to `port`, it takes type `string`.
-  constructor(public port: string) {
-    super(port);
+  constructor(public port: number) {
```

Comment:
You need to revert this to string if you want to capture ports that are not parsable as numbers.

Reply:
Ports are validated to be numbers at the API layer and can only be numbers in the domain layer. This here is not necessary, no more than other errors and their types.